docs/chapters/developer_guide.tex

```diff
@@ -0,0 +1,176 @@
+\chapter{Developer Guide}
+
+\section{Introduction}
+
+This chapter gives an overview of how \dfastmi can be setup for development.
+
+The development setup and configuration is done for usage of the visual studio code IDE. More information about visual studio code can be found on the internet.
+The development of python code project within Deltares is mostly done in with this tool. Visual Studio Code is free for private or commercial use. See the \href{https://code.visualstudio.com/license}{product license}) for details.
+
+\begin{Remark}
+	\item This configuration can not be used to build a standalone executable with nuitka tooling. It is for development purposes only.
+\end{Remark}
+
+\section{Configuration}
+The tool can be configured with a batch script which will configure using installed tools or ask the user to download and install the tools which can be used for them. The batch file is located in the sub folder \textbf{BuildScripts} and is called \textbf{DevelopDfastmi.bat}.
+
+The batch file will check for installed tooling for anaconda/miniconda. If not installed it will ask the user to install the anaconda/miniconda tooling for you. 
+\begin{Remark}
+	\item \textbf{PLEASE NOTE:} you need to close the command line and re-open / restart the batch script as it needs to initialize the conda tooling for the command line and powershell (used in VSCode).
+\end{Remark}
+
+The batch file will check for installed tooling for VSCode. If not installed it will ask the user to install the VSCode tooling for you. 
+
+\begin{Remark}
+	\item \textbf{PLEASE NOTE:} you need to close the command line and re-open / restart the batch script as it needs to initialize the VS Code tooling for the command line.
+\end{Remark}
+
+
+\section{Configuration files}
+As stated earlier we use conda to create a clean python environment to develop the \dfastmi application in. We also use poetry to setup the needed packages for the development environment. Poetry uses a toml\footnote{TOML : Tom's Obvious Minimal Language. A config file format file} file for the configuration (\textbf{poetry.toml}) and a lock file (\textbf{poetry.lock}) which i have no clue why it is there. Also we use a \textbf{setup.py} file used by the setup tools package to setup the application for development. And last but not least a python tools / interpreter configuration \textbf{pyproject.toml}. This file configures the way python packages should behave.
```

Comment:
Would you like to leave the following sentence in the developers guide?

```
A config file format file} file for the configuration (\textbf{poetry.toml}) and a lock file (\textbf{poetry.lock}) which i have no clue why it is there.
```

Reply:
If you want to keep this sentence, please use a capital "I"

Reply:
For more information on the poetry.lock file, see the following page:
https://python-poetry.org/docs/basic-usage/#installing-without-poetrylock

Reply:
I think i worded it now better.